docs/docs/reference/contextual/extension-methods.md

```diff
@@ -32,37 +34,68 @@ assert(circle.circumference == circumference(circle))
 
 ### Translation of Calls to Extension Methods
 
-When is an extension method applicable? There are two possibilities.
+When is an extension method applicable? There are two possibilities:
 
- - An extension method is applicable if it is visible under a simple name, by being defined
+ 1. An extension method is applicable if it is visible, by being defined
    or inherited or imported in a scope enclosing the application.
- - An extension method is applicable if it is a member of some given instance at the point of the application.
+ 2. An extension method is applicable if it is a member of some given instance at the point of the application.
 
-As an example, consider an extension method `longestStrings` on `Seq[String]` defined in a trait `StringSeqOps`.
+Following an example for the first point:
 
 ```scala
-trait StringSeqOps {
-  def (xs: Seq[String]).longestStrings = {
-    val maxLength = xs.map(_.length).max
-    xs.filter(_.length == maxLength)
-  }
+trait IntOps {
+  def (i: Int).isZero: Boolean = i == 0
+
+  def (i: Int).safeMod(x: Int): Option[Int] =
+    // extension method defined in same scope IntOps
+    if x.isZero then None
+    else Some(i % x)
+}
+
+object IntOpsEx extends IntOps {
+  def (i: Int).safeDiv(x: Int): Option[Int] =
+    // extension method brought into scope via inheritance from IntOps
+    if x.isZero then None
+    else Some(i % x)
+}
+
+trait SafeDiv {
+  import IntOpsEx._
+
+  def divide(i: Int, d: Int) : Option[(Int, Int)] =
+    // extension method imported and thus in scope
+    (i.safeDiv(d), i.safeMod(d)) match {
+      case (Some(d), Some(r)) => Some((d, r))
+      case _ => None
+    }
 }
 ```
-We can make the extension method available by defining a given `StringSeqOps` instance, like this:
+
+We build up on the above example to outline the second point.
+We can make an extension method available by defining a given instance containing it, like this:
 ```scala
-given ops1 as StringSeqOps
+given ops1 as IntOps // brings safeMod into scope
+
+1.safeMod(2)
 ```
-Then
+
+Then `safeMod` is legal everywhere `ops1` is available. Anonymous givens (and any other form of givens) are supported as well:
 ```scala
-List("here", "is", "a", "list").longestStrings
+given SafeDiv //brings safeMod, safeDiv and divide into scope
 ```
-is legal everywhere `ops1` is available. Alternatively, we can define `longestStrings` as a member of a normal object. But then the method has to be brought into scope to be usable as an extension method.
 
+In case the extension method is defined in an object as in `IntOpsEx`, then it can also be brought into scope by a given:
+```scala
+import given IntOpsEx.type = IntOpsEx // unusual, brings safeMod and safeDiv into scope
+```
+However importing from an object is most likely the better choice. And also superior regarding selectivety:
```

Comment:
Generally, it's better to have extension methods in givens (or rather, `extension`s, which maps to givens). So we should not advise people to use imports. That advice will be readily following since it's the conventional way to deal with extension methods, but it turns out it's an inferior way to model things.

Reply:
fair enough, I am going to remove this section but will wait until we get the `under a simple name` sorted.